utils/misc.cpp

```diff
@@ -468,68 +468,66 @@ link_sanity_check(void* link, int len, int precision, int dir, QudaGaugeParam* g
-	double* mylink = (double*)link;
-	//even
-	for (i = 0;i < len/2; i++){
-	    rc = link_sanity_check_internal_12(mylink + gaugeSiteSize*i, dir, i, gaugeParam, 0);
-	    if (rc != 0){
-		printf("ERROR: even link sanity check failed, i=%d\n",i);
-		display_link_internal(mylink+gaugeSiteSize*i);
-		exit(1);
-	    }
-	}
-	
-	mylink = mylink + gaugeSiteSize*len/2;
-	//odd
-	for (i = 0;i < len/2; i++){
-	    rc = link_sanity_check_internal_12(mylink + gaugeSiteSize*i, dir, i, gaugeParam, 1);
-	    if (rc != 0){
-		printf("ERROR: odd link sanity check failed, i=%d\n",i);
-		display_link_internal(mylink+gaugeSiteSize*i);
-		exit(1);
-	    }
-	}	
-	
+        double* mylink = (double*)link;
+        //even
+        for (i = 0;i < len/2; i++){
+            rc = link_sanity_check_internal_12(mylink + gauge_site_size*i, dir, i, gaugeParam, 0);
+            if (rc != 0){
+                printf("ERROR: even link sanity check failed, i=%d\n",i);
+                display_link_internal(mylink+gauge_site_size*i);
+                exit(1);
+            }
+        }
+
+        mylink = mylink + gauge_site_size*len/2;
+        //odd
+        for (i = 0;i < len/2; i++){
+            rc = link_sanity_check_internal_12(mylink + gauge_site_size*i, dir, i, gaugeParam, 1);
+            if (rc != 0){
+                printf("ERROR: odd link sanity check failed, i=%d\n",i);
+                display_link_internal(mylink+gauge_site_size*i);
+                exit(1);
+            }
+        }
+
     }else if (precision == QUDA_SINGLE_PRECISION){
-	float* mylink = (float*)link;
-
-	//even
-	for (i=0;i < len/2 ;i++){
-	    rc = link_sanity_check_internal_12(mylink + gaugeSiteSize*i, dir, i, gaugeParam, 0);
-	    if (rc != 0){
-		printf("ERROR: even link sanity check 12 failed, i=%d\n",i);
-		exit(1);
-	    }
-	    /*
-	    rc = link_sanity_check_internal_8(mylink + gaugeSiteSize*i, dir, i, gaugeParam, 0);
-	    if (rc != 0){
-		printf("ERROR: even link sanity check 8 failed, i=%d\n",i);
-		exit(1);
-	    }
-	    */
-	    
-	}
-	mylink = mylink + gaugeSiteSize*len/2;
-	//odd
-	for (i=0;i < len/2 ;i++){
-	    rc = link_sanity_check_internal_12(mylink + gaugeSiteSize*i, dir, i, gaugeParam, 1);
-	    if (rc != 0){
-		printf("ERROR: odd link sanity check 12 failed, i=%d\n", i);
-		exit(1);
-	    }	
-	    /*
-	    rc = link_sanity_check_internal_8(mylink + gaugeSiteSize*i, dir, i, gaugeParam, 0);
-	    if (rc != 0){
-		printf("ERROR: even link sanity check 8 failed, i=%d\n",i);
-		exit(1);
-	    }
-	    */
-	}	
+        float* mylink = (float*)link;
+
+        //even
+        for (i=0;i < len/2 ;i++){
+            rc = link_sanity_check_internal_12(mylink + gauge_site_size*i, dir, i, gaugeParam, 0);
+            if (rc != 0){
+                printf("ERROR: even link sanity check 12 failed, i=%d\n",i);
+                exit(1);
+            }
+
+            // rc = link_sanity_check_internal_8(mylink + gauge_site_size*i, dir, i, gaugeParam, 0);
```

Comment:
can we delete that?

Reply:
I couldn't see it being used anywhere in the tests, and nothing breaks when it's commented out. But they look like useful debugging routines.

Reply:
Indeed. But then we should use `#ifdef` I guess.

Reply:
Aye, I second using `#if 0` if anything for this

Reply:
Fixed (deleted since it's dead code).